src/views/List.jsx

```diff
@@ -1,13 +1,50 @@
+import { useEffect, useState } from 'react';
 import { ListItem } from '../components';
 
 export function List({ data, listToken }) {
+	const [searchTerm, setSearchTerm] = useState('');
+	const [filteredData, setFilteredData] = useState([]);
+
+	// Function to update the filteredData based on the searchTerm
+	const updateFilteredData = () => {
+		const filteredItems = data.filter((item) => {
+			if (item.name) {
+				return item.name.toLowerCase().includes(searchTerm.toLowerCase());
+			}
+		});
+		setFilteredData(filteredItems);
+	};
+
+	// Update filteredData whenever searchTerm or data changes
+	useEffect(() => {
+		updateFilteredData();
+	}, [searchTerm, data]);
+
+	const handleClear = () => {
+		setSearchTerm('');
+		setFilteredData(data);
+	};
+
 	return (
 		<>
-			<p>
-				Hello from the <code>/list</code> page!
-			</p>
+			<form className="filter-form">
+				<label htmlFor="search-term">Filter items:</label>
+				<input
```

Comment:
It's not an A/C, so more just something to think about, but from an accessibility standpoint, it's worth considering how we might also make it easier to clear the search field with just a key press.